pkg/app/mongodb.go

```diff
@@ -0,0 +1,220 @@
+	if err != nil {
+		return err
+	}
+
+	log.Print("Installing application using helm.", field.M{"app": mongo.name})
+	err = cli.Install(ctx, fmt.Sprintf("%s/%s", mongo.chart.RepoName, mongo.chart.Chart), mongo.chart.Version, mongo.name, mongo.namespace, mongo.chart.Values)
+
+	if err != nil {
+		return err
+	}
+	log.Print("Application was installed successfully.", field.M{"app": mongo.name})
+	return nil
+}
+
+func (mongo *MongoDB) IsReady(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+	defer cancel()
+
+	err := kube.WaitOnStatefulSetReady(ctx, mongo.cli, mongo.namespace, fmt.Sprintf("%s-mongodb-primary", mongo.name))
+	if err != nil {
+		return false, err
+	}
+
+	err = kube.WaitOnStatefulSetReady(ctx, mongo.cli, mongo.namespace, fmt.Sprintf("%s-mongodb-secondary", mongo.name))
+	if err != nil {
+		return false, err
+	}
+
+	err = kube.WaitOnStatefulSetReady(ctx, mongo.cli, mongo.namespace, fmt.Sprintf("%s-mongodb-arbiter", mongo.name))
+	if err != nil {
+		return false, err
+	}
+
+	log.Print("Application is ready.", field.M{"app": mongo.name})
+	return true, nil
+}
+
+func (mongo *MongoDB) Object() crv1alpha1.ObjectReference {
+	return crv1alpha1.ObjectReference{
+		Kind:      "StatefulSet",
+		Name:      fmt.Sprintf("%s-mongodb-primary", mongo.name),
+		Namespace: mongo.namespace,
+	}
+}
+
+func (mongo *MongoDB) Uninstall(ctx context.Context) error {
+	cli := helm.NewCliClient(helm.V3)
+
+	log.Print("Uninstalling application.", field.M{"app": mongo.name})
+	err := cli.Uninstall(ctx, mongo.name, mongo.namespace)
+	if err != nil {
+		return errors.Wrapf(err, "Error while uninstalling the application.")
+	}
+
+	return nil
+}
+
+func (mongo *MongoDB) Ping(ctx context.Context) error {
+	log.Print("Pinging the application.", field.M{"app": mongo.name})
+	pingCMD := []string{"sh", "-c", fmt.Sprintf("mongo admin --authenticationDatabase admin -u %s -p $MONGODB_ROOT_PASSWORD --quiet --eval \"rs.slaveOk(); db\"", mongo.username)}
+	_, stderr, err := mongo.execCommand(ctx, pingCMD)
+	if err != nil {
+		return errors.Wrapf(err, "Error while pinging the mongodb application %s", stderr)
+	}
+
+	// even after ping is successful, it takes some time for primary pod to becomd the master
+	// we will have to wait for that so that the write subsequent write requests wont fail.
+	isMasterCMD := []string{"sh", "-c", fmt.Sprintf(" mongo admin --authenticationDatabase admin -u %s -p $MONGODB_ROOT_PASSWORD --quiet --eval \"db.isMaster()\"", mongo.username)}
+	isMasterStdout, stderr, err := mongo.execCommand(ctx, isMasterCMD)
+	if err != nil {
+		return errors.Wrapf(err, "Error %s checking if the pod is master.", stderr)
+	}
+
+	// extract the ismaster field from the output and make sure
+	// this ismaster field is true
+	re := regexp.MustCompile("\"ismaster\" : ([a-z]*),")
+	match := re.FindStringSubmatch(isMasterStdout)
+
+	isMaster, err := strconv.ParseBool(match[1])
```

Comment:
We should ensure `len(match) >= 2` before accessing the element

Reply:
So, output of `db.ismaster` is always going to have `ismaster` field in output so we are always going to have `len(match)>= 2`. But yeah, I think we should that.